integration-tests/docker/docker-compose.base.yml

```diff
@@ -0,0 +1,271 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```

Comment:
Is it possible to use the environment variables we define in the scripts for the path to mount volumes and env files here. (i.e.  COMMON_ENV, SERVICE_SUPERVISORDS_DIR, SUPERVISORDIR, RESOURCEDIR, etc.)

Reply:
yes it's possible, we can create file with env variables, but then this docker-compose file will be bad readable.
What is the reason to use variable like: "COMMON_ENV"?
Env variables used in case if you want to customize something

------- Currently -------
This part is readable, you can understand that we mount file "zookeeper.conf" from "current dir/service-supervisords/zookeeper.conf" to some path in your container and you can see this path
    volumes:
      - ${HOME}/shared:/shared
      - ./service-supervisords/zookeeper.conf:/usr/lib/druid/conf/zookeeper.conf
      - ./service-supervisords/kafka.conf:/usr/lib/druid/conf/kafka.conf
------- With env variables like in run_cluster.sh -------
look at this configuration from person who is new. What he see? He see a lot of env variables, he don't know what the values of this variables, he need to spend some time to find where this variables declared to see path. This configuration is hard to understand 
    volumes:
      - ${HOME}/shared:/shared
      - ${SERVICE_SUPERVISORDS_DIR}/zookeeper.conf:${SUPERVISORDIR}/zookeeper.conf
      - ${SERVICE_SUPERVISORDS_DIR}/kafka.conf:${SUPERVISORDIR}/kafka.conf

Reply:
I think this is to keep it in sync with the other scripts. For example, if I move the folder path in build_run_cluster.sh to something different then the path will be wrong in this file.

Reply:
you need to split "env variables for customizing server" and "env variables that you created for development".
We need to avoid "env variable for development" in docker-compose files.
If you have some configuration for your continer this config files must be in close folder.
If you want to move docker compose files you need to move then with configuration files. they need to be in the same folder because they have the same context

Folders paths, configs paths, any other things that related to container config they must be in "docker-compose" files, not in shell scripts, java pom.xml, etc

Reply:
Should the SHARED_DIR be passed to yml the same way as OVERRIDE_ENV (instead of hardcoding it to ${HOME}/shared). Since SHARED_DIR does not have to be in the same directory as the yml and can be anywhere.

Reply:
Do we have different configurations for shared_dir path?

Reply:
Right now it is hardcoded to otuside of Druid directory at ${HOME}/shared

Reply:
Currently it is done in correct way.
All paths that related to server configuration that need to be done in yaml and you dont need to create env variable for this.